modules/jolt_physics/objects/jolt_area_3d.cpp

```diff
@@ -197,82 +197,13 @@ void JoltArea3D::_notify_body_exited(const JPH::BodyID &p_body_id) {
 	}
 }
 
-void JoltArea3D::_force_bodies_entered() {
+void JoltArea3D::_remove_all_overlaps() {
 	for (KeyValue<JPH::BodyID, Overlap> &E : bodies_by_id) {
-		Overlap &body = E.value;
-
-		if (unlikely(body.shape_pairs.is_empty())) {
-			continue;
-		}
-
-		for (const KeyValue<ShapeIDPair, ShapeIndexPair> &P : body.shape_pairs) {
-			body.pending_removed.erase(P.value);
-			body.pending_added.push_back(P.value);
-		}
-
-		_events_changed();
-	}
-}
-
-void JoltArea3D::_force_bodies_exited(bool p_remove) {
-	for (KeyValue<JPH::BodyID, Overlap> &E : bodies_by_id) {
-		const JPH::BodyID &id = E.key;
-		Overlap &body = E.value;
-
-		if (unlikely(body.shape_pairs.is_empty())) {
-			continue;
-		}
-
-		for (const KeyValue<ShapeIDPair, ShapeIndexPair> &P : body.shape_pairs) {
-			body.pending_added.erase(P.value);
-			body.pending_removed.push_back(P.value);
-		}
-
-		_events_changed();
-
-		if (p_remove) {
-			body.shape_pairs.clear();
-			_notify_body_exited(id);
-		}
-	}
-}
-
-void JoltArea3D::_force_areas_entered() {
-	for (KeyValue<JPH::BodyID, Overlap> &E : areas_by_id) {
-		Overlap &area = E.value;
-
-		if (unlikely(area.shape_pairs.is_empty())) {
-			continue;
-		}
-
-		for (const KeyValue<ShapeIDPair, ShapeIndexPair> &P : area.shape_pairs) {
-			area.pending_removed.erase(P.value);
-			area.pending_added.push_back(P.value);
-		}
-
-		_events_changed();
+		_notify_body_exited(E.key);
 	}
-}
-
-void JoltArea3D::_force_areas_exited(bool p_remove) {
-	for (KeyValue<JPH::BodyID, Overlap> &E : areas_by_id) {
-		Overlap &area = E.value;
-
-		if (unlikely(area.shape_pairs.is_empty())) {
-			continue;
-		}
 
-		for (const KeyValue<ShapeIDPair, ShapeIndexPair> &P : area.shape_pairs) {
-			area.pending_added.erase(P.value);
-			area.pending_removed.push_back(P.value);
-		}
-
-		_events_changed();
-
-		if (p_remove) {
-			area.shape_pairs.clear();
-		}
-	}
+	bodies_by_id.clear();
+	areas_by_id.clear();
```

Comment:
Just checking: Should this not send notifications to `body_monitor_callback` and `area_monitor_callback`?

Reply:
No, I thought so too, but since the scene node (i.e. the receiver of the callback) is going away anyway, or worse yet, is being replaced by another scene node, any calls to that callback will either not happen (because nulled out callback) or will happen to the wrong scene node.

So we should rightfully just wipe everything in this manner instead.